portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -126,4 +143,42 @@ private static String convertToJsonUsingGson(ArrayList<Comment> list) {
         Gson gson = new Gson();
         return gson.toJson(list);
     }
+
+    /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
+    private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+        BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+        Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+        List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+        // User submitted form without selecting a file, so we can't get a URL. (dev server)
+        if (blobKeys == null || blobKeys.isEmpty()) {
+            return null;
+        }
+
+        // Our form only contains a single file input, so get the first index.
```

Comment:
Can you add a check to make sure this holds true?

Reply:
done